File: python/Q191_number_of_1_bits.py
#
# @lc app=leetcode id=191 lang=python3
#
# [191] Number of 1 Bits
#

# @lc code=start
class Solution:
    def hammingWeight(self, n: int) -> int:
        result: int = 0

        while n != 0:
            n &= n - 1
            result += 1

        return result


# @lc code=end

File: rust/src/leetcode/Q191_number_of_1_bits.rs
/*
 * @lc app=leetcode id=191 lang=rust
 *
 * [191] Number of 1 Bits
 */

// @lc code=start
impl Solution {
    pub fn hamming_weight(n: i32) -> i32 {
        let mut result: i32 = 0;
        let mut number: i32 = n;

        while number != 0 {
            result += number & 1;
            number >>= 1;
        }

        result
    }
}
// @lc code=end
